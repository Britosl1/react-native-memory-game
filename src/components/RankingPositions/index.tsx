import React from 'react';
import {View, StyleSheet, Text} from 'react-native';

interface IRankingPositionsProps {
  name: string;
  score: number;
}

const styles = StyleSheet.create({
  container: {
    borderRadius: 3,
    height: 30,
    backgroundColor: '#3c1361',
    justifyContent: 'space-between',
    alignItems: 'center',
    paddingHorizontal: 20,
    flexDirection: 'row',
  },
  text: {
    color: 'white',
    fontWeight: '900',
    fontSize: 14,
  },
});

const RankingPositions: React.FC<IRankingPositionsProps> = ({name, score}) => {
  return (
    <View style={styles.container}>
      <Text style={styles.text}>{`Name: ${name}`}</Text>
      <Text style={styles.text}>{`Score: ${score}`}</Text>
    </View>
  );
};

export default RankingPositions;
