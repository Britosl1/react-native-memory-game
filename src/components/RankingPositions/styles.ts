import styled from 'styled-components/native';

export const RankingContainer = styled.View`
  border-radius: 3px;
  height: 30px;
  background-color: #3c1361;
  justify-content: space-between;
  align-items: center;
  padding-left: 20px;
  padding-right: 20px;
  flex-direction: row;
  margin-bottom: 10px;
`;

export const TextRankingContainer = styled.Text`
  color: white;
  font-weight: 900;
  font-size: 14px;
`;
