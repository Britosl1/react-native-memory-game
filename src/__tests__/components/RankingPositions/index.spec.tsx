import React from 'react';
import {render} from '@testing-library/react-native';
import RankingPositions from '../../../components/RankingPositions';

describe('Snapshot test', () => {
  test('Ranking component', () => {
    const {toJSON} = render(<RankingPositions name="Lucas" score="2" />);

    expect(toJSON()).toMatchSnapshot();
  });
});
