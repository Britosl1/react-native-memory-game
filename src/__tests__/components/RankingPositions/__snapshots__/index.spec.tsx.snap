// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test Ranking component 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#3c1361",
        "borderRadius": 3,
        "flexDirection": "row",
        "height": 30,
        "justifyContent": "space-between",
        "marginBottom": 10,
        "paddingLeft": 20,
        "paddingRight": 20,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "white",
          "fontSize": 14,
          "fontWeight": "900",
        },
      ]
    }
  >
    Name: Lucas
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "white",
          "fontSize": 14,
          "fontWeight": "900",
        },
      ]
    }
  >
    Score: 2
  </Text>
</View>
`;
